CMakeLists.txt

```diff
@@ -1,16 +1,44 @@
 cmake_minimum_required (VERSION 3.4)
+
+include(arm-none-eabi.cmake)
+
 project (ST-LIB)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-set(LIBRARY st-lib)
 
+set(LIBRARY st-lib)
+  
 file(GLOB_RECURSE SOURCE_C ${CMAKE_SOURCE_DIR}/ *.c)
```

Comment:
This is wrong. As I mentioned you in person, there is the space here, remove it for now so it actually matches the correct regex. I recommend you to remove the regex and manually input all the files. Or even better as you moved it to python you could dinamically find the source files using a regex in python and modify this file from there. The difference in using a python regex is that you can create you own feedback output, while GLOB does not output the regex.

Reply:
I understand what you are saying but look at this, 
on the right there is the build removing the whitespaces, on the left the build with whitespaces, it does not compile any more files.
![image](https://github.com/HyperloopUPV-H8/ST-LIB/assets/58850783/db8389f6-a1e9-40aa-b451-bafcfd2468ee)

 Do you think if there were extra .cpp files it would compile them?
It does not search for all of the cpp files, NOTE that this happens with the following line: `file(GLOB_RECURSE SOURCE_CPP ${CMAKE_SOURCE_DIR}/ /*.cpp)` with the second slash it is clear that it searches starting from / but in the case of the whitespace seems to ignore it

However I agree that the GLOB approach is not ideal and I will manually add the files.

Reply:
Very interesting, did not know :+1: